cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go

```diff
@@ -253,6 +253,50 @@ func AvailablePackageSummaryFromUnstructured(pkgMeta *unstructured.Unstructured,
 	return summary, nil
 }
 
+// GetAvailablePackageVersions returns the package versions managed by the 'helm' plugin
+func (s *Server) GetAvailablePackageVersions(ctx context.Context, request *corev1.GetAvailablePackageVersionsRequest) (*corev1.GetAvailablePackageVersionsResponse, error) {
+	namespace := request.GetAvailablePackageRef().GetContext().GetNamespace()
+	cluster := request.GetAvailablePackageRef().GetContext().GetCluster()
+	log.Infof("+kapp-controller GetAvailablePackageVersions (cluster=%q, namespace=%q)", cluster, namespace)
+
+	identifier := request.GetAvailablePackageRef().GetIdentifier()
+	if namespace == "" || identifier == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Required context or identifier not provided")
+	}
+	log.Errorf("request available pkg ref was: %+v", request.GetAvailablePackageRef())
+
+	client, err := s.getDynamicClient(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use the field selector to return only Package CRs that match on the spec.refName.
+	pkgGVR := schema.GroupVersionResource{Group: packageGroup, Version: packageVersion, Resource: packageResources}
+	pkgs, err := client.Resource(pkgGVR).Namespace(namespace).List(ctx, metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.refName=%s", identifier),
```

Comment:
Ah great, now I get this slack msg (https://kubernetes.slack.com/archives/CH8KCCKA5/p1636331168315300?thread_ts=1636331095.314400&cid=CH8KCCKA5)